### DISCUSSION TITLE
should `Rack::ContentLength` close the body?

### DISCUSSION THREAD

**Discussion by lygaret** [rack/rack#2111] [Q&A] [answered]
Hey, I'm doing this as a discussion rather than an issue because I can't imagine I'm doing this right:

In `Rack::ContentLength`, the call method calls `#to_ary` on the body, but never calls `#close`.
Because of this, anything that uses the BodyProxy to run a block when the request is fully "sent" isn't getting done; I found it by noticing missing log lines from `CommonLogger`.

This seems like something that would screw other folk up though, so I'm curious, how are y'all handling that? _Should_ the `ContentLength` middleware be calling `#close`? Is this just a middleware ordering problem on my end?

I've copied the middleware out to my own project and have added:

```ruby
body.close if body.respond_to(:close)
```

just _after_ the body is serialized, and it solved my issue -- is this worth submitting a PR for?

Thanks!

**Comment by lygaret** [comment]
Just found this:
https://github.com/rack/rack/pull/2062

Which makes me think that I'm just out of date...

**Reply by jeremyevans** [reply, marked as answer]
It's fixed in master, but it is not yet in any release.

**Reply by lygaret** [reply]
Perfect, thank you. I thought I was crazy.
